docs/developers/cross-chain-messaging/examples/cross-chain-counter.md

```diff
@@ -27,8 +27,140 @@ yarn add --dev @openzeppelin/contracts
 
 ## Create a new contract
 
-```solidity title="contracts/CrossChainCounter.sol" reference
-https://github.com/zeta-chain/example-contracts/blob/3e79d006239bb9cbe0120e36aecc471cea4a2ad7/messaging/counter/contracts/CrossChainCounter.sol
+```
+npx hardhat messaging CrossChainCounter messageFrom:address
+```
+
+- `messageFrom`: address of the sender
+
+```solidity title="contracts/CrossChainCounter.sol"
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.7;
+
+import "@openzeppelin/contracts/interfaces/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@zetachain/protocol-contracts/contracts/evm/tools/ZetaInteractor.sol";
+import "@zetachain/protocol-contracts/contracts/evm/interfaces/ZetaInterfaces.sol";
+
+interface CrossChainCounterErrors {
+    error InvalidMessageType();
+
+    // highlight-next-line
+    error DecrementOverflow();
+}
+
+contract CrossChainCounter is
+    ZetaInteractor,
+    ZetaReceiver,
+    CrossChainCounterErrors
+{
+    // highlight-start
+    bytes32 public constant CROSS_CHAIN_INCREMENT_MESSAGE =
```

Comment:
Let's make as few changes as possible to streamline the developer experience. There is only one message type, we might as well keep it the way it was generated by the `messaging` command: `CROSS_CHAIN_COUNTER_MESSAGE_TYPE`.

Reply:
Ok, I didn't realize it generates `CROSS_CHAIN_CROSS_CHAIN_COUNTER`, which is not pretty. We might want to change the template in the future to be less verbose, but for now I think it makes sense to just switch to:

```
npx hardhat messaging counter from:address
```